src/Content/Conversation/Repository/UserDefinedChannel.php

```diff
@@ -221,62 +219,53 @@ public function getMatchingChannelUsers(string $searchtext, string $language, ar
 			return [];
 		}
 
-		$this->insertCheckFullTextSearch($searchtext);
+		$disposableFullTextSearch = new DisposableFullTextSearch($this->db, $searchtext);
 
-		$uids = [];
+		$filteredChannels = $this->select(['uid' => array_column($users, 'uid'), 'publish' => true, 'valid' => true])->filter(
```

Comment:
What is this `filter` function and how exactly does it work?

Reply:
Filter returns a filtered copy of the array argument depending on the return value of a callback that is performed on each value. If the return value is truthy, the value is kept, if the value is falsy, the value is discarded from the returned array.